### PR TITLE
Create stepper component

### PR DESCRIPTION
**Pull request by przemyslawjanpietrzak**

### CODE REVIEW

File: lib/js/atoms/Stepper.vue [hunk 1]
```diff
@@ -0,0 +1,117 @@
+<template>
+	<div class="stepper">
+		<div v-for="(step, index) in steps" :key="index" class="stepper__item">
+			<div
+				v-if="index !== 0"
+				class="stepper__separator"
+				:class="{ '-active': step.isActive }"
+			></div>
+			<div class="stepper__icon" :class="{ '-active': step.isActive }">
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:9]: Is it intended that multiple steps are active at the same time? 
Maybe we should consider having only one step active, all steps before that step should be also marked green. Such encapsulation avoids leaking isActive logic to the application

File: lib/js/atoms/Stepper.vue [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+				<icon :icon="ICONS[step.iconKey]" :size="ICON_SIZES.X_SMALL"></icon>
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:10]: On designs I see different icon for steps that are finished. How do we want to address that?

File: lib/js/atoms/Stepper.vue [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		}
+	}
+
+	&__icon {
+		align-self: center;
+		background-color: $color-minor-background;
+		border-radius: 100%;
+		color: $color-minor;
+		display: flex;
+		height: 32px;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:51]: variables

File: lib/js/atoms/Stepper.vue [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		@include textBold();
+
+		color: $color-minor;
+
+		&.-active {
+			color: $color-primary;
+		}
+	}
+
+	&__icon {
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:45]: do we need additional wrapper on icon?
Maybe we could apply styles on `<wnl-icon class="stepper__icon">`? This way you could avoid setting div size

File: lib/js/atoms/Stepper.vue [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+		transition: background-color 0.33s;
+
+		&.-active {
+			background-color: $color-primary-background;
+			color: $color-primary;
+		}
+	}
+
+	&__separator {
+		background-color: #e6e8ee;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:64]: variables

File: lib/js/atoms/Stepper.vue [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+		border-radius: 2.5px;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:65]: please discuss with UX if we really want to use decimals here

File: lib/js/atoms/Stepper.vue [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+		height: 2px;
+		margin: 0 $space-xs;
+		transition: background-color 0.33s;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:68]: I see `0.33s` multiple times. It's a nice candiadate for a variable. Can be local for now...

File: lib/js/atoms/Stepper.vue [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+		width: 32px;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:69]: it shouldn't be fixed width. We can define mix and max, but definitely it should be adjusting based on the number of steps and window size

File: lib/js/atoms/Stepper.vue [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+	props: {
+		steps: {
+			type: Array as Prop<Array<Step>>,
+			required: true,
+			validate: (steps: Array<Step>) => {
+				return steps.every((step) => {
+					if (typeof step.label !== 'string') {
+						return false;
+					}
+					if (typeof step.iconKey !== 'string') {
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:100]: probably we should validate if Icon with this name exists

File: lib/js/atoms/Stepper.vue [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+export default {
+	name: 'Stepper',
+	components: { Icon },
+	props: {
+		steps: {
+			type: Array as Prop<Array<Step>>,
+			required: true,
+			validate: (steps: Array<Step>) => {
+				return steps.every((step) => {
+					if (typeof step.label !== 'string') {
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:97]: won't TS validate that? Do we need to do that manually?

File: lib/js/atoms/Stepper.vue [hunk 11]
```diff
@@ -0,0 +1,117 @@
...
+@import '../../styles/settings/media-queries';
+@import '../../styles/settings/typography';
+
+.stepper {
+	display: flex;
+
+	&__item {
+		align-items: center;
+		display: flex;
+		justify-content: center;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:31]: On mobile label is under the icon on designs

File: lib/js/atoms/Stepper.vue (hunk 1 above)
Comment (przemyslawjanpietrzak) [reply lib/js/atoms/Stepper.vue:9]: I thought about it, however I choose this solution

motivation:
 - this is DS component so should be as reusable and logic agnostic as possible
 - on the application level we gonna to write wrapper for applying that logic (if stepper will be only application component this 2 components should be merged)

File: lib/js/atoms/Stepper.vue (hunk 2 above)
Comment (przemyslawjanpietrzak) [reply lib/js/atoms/Stepper.vue:10]: as above, in my opinion it's application specific logic, so it should not be in DS

File: lib/js/atoms/Stepper.vue (hunk 7 above)
Comment (przemyslawjanpietrzak) [reply lib/js/atoms/Stepper.vue:68]: Agree, but let's keep in `styles/settings`, ok?

File: lib/js/atoms/Stepper.vue (hunk 9 above)
Comment (przemyslawjanpietrzak) [reply lib/js/atoms/Stepper.vue:100]: we do that, but in a child component

File: lib/js/atoms/Stepper.vue (hunk 10 above)
Comment (przemyslawjanpietrzak) [reply lib/js/atoms/Stepper.vue:97]: TS doesn't works on runtime, validator works. Also TS doesn't check props applied to rendered component.

I'm not fan of validators (e.g. react with TS doesn't validate props in runtime, however JSX allows to make that check in compilation time.

File: lib/js/atoms/Stepper.vue [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+			width: 151px;
+		}
+
+		&.-filled {
+			background-color: $color-primary;
+		}
+	}
+
+	&__ripple {
+		height: $step-icon-size;
```
[the remark this answers is not in the document]
Comment (kvas-damian) [reply lib/js/atoms/Stepper.vue:105]: but then ripple is too small :(

File: lib/js/atoms/Stepper.vue (hunk 8 above)
Comment (przemyslawjanpietrzak) [reply lib/js/atoms/Stepper.vue:93]: Ewa is gonna to send us exact requirements, however we don't need it right now

File: lib/js/atoms/Stepper.types.ts [hunk 13]
```diff
@@ -0,0 +1,7 @@
+export interface Step {
+	label: string;
+	iconKey: string;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.types.ts:3]: Passing `iconKey` can cause some trubble here...
You need to hardcode string, or hack the map of icons to get the key. What do you think about passing icon object instead? Link in `Button` component?

File: lib/js/atoms/Stepper.vue (hunk 11 above)
Comment (kvas-damian) [reply lib/js/atoms/Stepper.vue]: @przemyslawjanpietrzak any update here. I still see it's not fixed

File: lib/js/atoms/Stepper.vue [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+	created() {
+		this.ICONS = ICONS;
+		this.ICON_SIZES = ICON_SIZES;
+	},
+	methods: {
+		onStepClick(step: Step) {
+			if (!step.isClickable) {
+				return;
+			}
+			this.$emit('click', step.route);
```
[the remark this answers is not in the document]
Comment (kvas-damian) [reply lib/js/atoms/Stepper.vue]: What about event name? Do we want to address that?

File: lib/js/atoms/Stepper.vue [hunk 15]
```diff
@@ -0,0 +1,152 @@
...
+		margin-right: $space-xxs;
+		width: $step-icon-size;
+		transition: background-color $basic-transition-time;
+
+		&.-filled {
+			background-color: $color-primary-background;
+			color: $color-primary;
+
+			&:hover:not(.-disabled) {
+				background-color: $color-primary-background-hovered;
```
Comment (kvas-damian) [inline lib/js/atoms/Stepper.vue:74]: This color looks different in Invision. Can we double-check that?

Comment (przemyslawjanpietrzak) [reply lib/js/atoms/Stepper.vue:93]: UX has no proposition how colors should be calculated, they agreed to stay with mix of 16% firefly-black

**Pull request closed without merging**